frontend/ui/widget/dictquicklookup.lua

```diff
@@ -1169,6 +1176,11 @@ function DictQuickLookup:onHoldClose(no_clear)
     return true
 end
 
+function DictQuickLookup:onDictQuickLookupToHome()
+    self:onHoldClose()
+    self.ui:onHome()
+end
```

Comment:
I've seen that you added this kind of thing in some other PR.
I understand this brings you back to File browser ? If that's right, it feels a bit hardcore effect, nearly like exiting the whole thing on a single key press which may be done easily by error.
Is this really necessary ? Shouldn't it just quit any popup and brings you back to Reader, like Esc/Back ?

Reply:
> I've seen that you added this kind of thing in some other PR.
I understand this brings you back to File browser ? 

Yes

> If that's right, it feels a bit hardcore effect, nearly like exiting the whole thing on a single key press which may be done easily by error.

Do you mean, the normal thing that every home button is supposed to do? It's a home button, it needs to take you back home, wherever you are. So yes, i want to properly support it.

Reply:
> Do you mean, the normal thing that every home button is supposed to do? ;) It's a home button, it needs to take you back home, wherever you are. 

I understand. But I don't consider File browser to be (my) Home. File browser is just a utility where I go choose my next book, not something I consider a home.
If deep in dict lookups, I would accept Home to bring me back quickly into the book, which is my current home.
We don't have a notion of "home" (even if there is a :onHome in Reader). What we have is just a user choice of "Start with". Dunno if this would be preferable.

Also, I don't know if it's sane to add everywhere `:onDictQuickLookupToHome() self:onHoldClose()    self.ui:onHome() end`. Doesn't just sending "Home" reaches Reader and take care of closing every popup? (no idea, just asking).

Reply:
You can also use the dictionary in the file browser btw
 What does it do there exactly, also leave the folder you're in at the same time?

Reply:
Well one could say, home is where all the books live i.e file browser’s home directory. File Manager also has an `onHome` that takes you to the home directory btw.

> Also, I don't know if it's sane to add everywhere :onDictQuickLookupToHome() 

It isn’t, each widget needs their own version of it. This one works just for dictionary and yes, it closes all instances of the dictionary (`onHoldClose` does, `onHome` does not close anything besides reader). Same with the page browser and book map, they also close all their own instances.

I still need to write all the other ones, but one i can’t seem to figure out is the TouchMenu one where the `self.ui:onHome` crashes the whole thing. Any ideas?

Reply:
> You can also use the dictionary in the file browser btw What does it do there exactly, also leave the folder you're in at the same time?

Yes, puts you in the home directory when triggered from file browser. if inside reader, it puts you in the folder where the last open book is.

Reply:
I have to agree that sounds quite aggressive.

Reply:
But that is what home buttons do...

Reply:
I suppose, though on Android for example it doesn't close anything immediately and recovering from a mistake only takes a second or two. But that aside. Perhaps it doesn't matter too much atm since most devices don't have home buttons, but then again I'd rather get it right.

As @poire-z says, I added the "starts with" setting to provide a choice of "home," which isn't quite the same thing as the home folder.

Reply:
> but then again I'd rather get it right.

By "getting it right" you mean allowing users to select whether home means FM or Favourites or History???

Reply:
I mainly mean thinking about what getting it right actually means but that does seem like the primary candidate.

Reply:
What about an option to select either back to reader or back to home folder?

```lua
function onSomewhereToHome
    self:onClose
    if G_setting(back_to_home_folder)
        self.ui:onHome
    end
end
```

Reply:
To my mind it would simply use the existing `G_reader_settings:readSetting("start_with")` setting, with the caveat that it should go to the file browser if the setting is last document and not just if it's not set.

https://github.com/koreader/koreader/blob/4cdef40f9720006aecb96120b8018cd917ecaf6f/reader.lua#L219